processing/src/main/java/io/druid/query/aggregation/DoubleMaxAggregatorFactory.java

```diff
@@ -52,25 +54,43 @@ public DoubleMaxAggregatorFactory(String name, String fieldName)
   }
 
   @Override
-  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  protected ColumnValueSelector selector(ColumnSelectorFactory metricFactory)
   {
-    return new DoubleMaxAggregator(getDoubleColumnSelector(metricFactory, Double.NEGATIVE_INFINITY));
+    return getDoubleColumnSelector(
+        metricFactory,
+        Double.NEGATIVE_INFINITY
+    );
   }
 
   @Override
-  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  protected Aggregator factorize(ColumnSelectorFactory metricFactory, ColumnValueSelector selector)
   {
-    return new DoubleMaxBufferAggregator(getDoubleColumnSelector(metricFactory, Double.NEGATIVE_INFINITY));
+    return new DoubleMaxAggregator(selector);
   }
 
   @Override
-  public Object combine(Object lhs, Object rhs)
+  protected BufferAggregator factorizeBuffered(
+      ColumnSelectorFactory metricFactory, ColumnValueSelector selector
```

Comment:
Please format either the whole signature in one line, or each param on a separate string. Same below in Factories.

Reply:
Will do. 
Can you add a checkstyle rule that fails when this is not done ?

Reply:
done.

Reply:
No, when I checked the last time there was no way to enforce it with checkstyle.